sync/class.jetpack-sync-module-users.php

```diff
@@ -157,6 +157,41 @@ function save_user_cap_handler( $meta_id, $user_id, $meta_key, $capabilities ) {
 		}
 	}
 
+	protected function enqueue_all_ids_as_action( $action_name, $table_name, $id_field, $where_sql ) {
+		global $wpdb;
+
+		if ( ! $where_sql ) {
+			$where_sql = '1 = 1';
+		}
+
+		$items_per_page = 1000;
+		$page           = 1;
+		$chunk_count    = 0;
+		$previous_id    = 0;
+
+		while ( $ids = $wpdb->get_col( "SELECT user_id FROM $wpdb->usermeta WHERE meta_key = '{$wpdb->base_prefix}user_level' AND {$where_sql} AND user_id > {$previous_id} ORDER BY user_id ASC LIMIT {$items_per_page}" ) ) {
+			// Request posts in groups of N for efficiency
+			$chunked_ids = array_chunk( $ids, self::ARRAY_CHUNK_SIZE );
+
+			// Send each chunk as an array of objects
+			foreach ( $chunked_ids as $chunk ) {
+				/**
+				 * Fires with a chunk of object IDs during full sync.
+				 * These are expanded to full objects before upload
+				 *
+				 * @since 4.2.0
+				 */
+				do_action( $action_name, $chunk );
```

Comment:
I'm not 100% sure how the doc blocks work, but I anticipate that we'll need to change this so we're passing an actual string to `do_action()`. cc @jeherve